modules/ml/src/main/java/org/apache/ignite/ml/svm/SVMLinearBinaryClassificationTrainer.java

```diff
@@ -67,34 +87,32 @@
 
     /** */
     @NotNull private Vector initializeWeightsWithZeros(int vectorSize) {
-        if (isDistributed)
-            return new SparseDistributedVector(vectorSize);
-        else
             return new DenseLocalOnHeapVector(vectorSize);
     }
 
     /** */
-    private Vector calculateUpdates(LabeledDataset data, Vector weights) {
-        Vector copiedWeights = weights.copy();
-        Vector deltaWeights = initializeWeightsWithZeros(weights.size());
-
-        final int amountOfObservation = data.rowSize();
+    private Vector calculateUpdates(Vector weights) {
+        return dataset.compute(data -> {
+            Vector copiedWeights = weights.copy();
+            Vector deltaWeights = initializeWeightsWithZeros(weights.size());
+            final int amountOfObservation = data.rowSize();
 
-        Vector tmpAlphas = initializeWeightsWithZeros(amountOfObservation);
-        Vector deltaAlphas = initializeWeightsWithZeros(amountOfObservation);
+            Vector tmpAlphas = initializeWeightsWithZeros(amountOfObservation);
+            Vector deltaAlphas = initializeWeightsWithZeros(amountOfObservation);
 
-        for (int i = 0; i < this.getAmountOfLocIterations(); i++) {
-            int randomIdx = ThreadLocalRandom.current().nextInt(amountOfObservation);
+            for (int i = 0; i < this.getAmountOfLocIterations(); i++) {
```

Comment:
It looks like due to  'this.getAmountOfLocIterations()' the whole instance of the trainer will be serialized every time this lambda will be sent anyway. Am I right? The same question about `initializeWeightsWithZeros` method calls.

Bye the way, it might be not a problem if we remove field with dataset. But in this case we need to mark the class as `Serializable` anyway.